cmd/d2/watch.go

```diff
@@ -325,7 +329,20 @@ func (w *watcher) compileLoop(ctx context.Context) error {
 			recompiledPrefix = "re"
 		}
 
-		b, err := compile(ctx, w.ms, w.layoutPlugin, w.inputPath, w.outputPath)
+		if filepath.Ext(w.outputPath) == ".png" && !w.pw.Browser.IsConnected() {
+			newPW, err := w.pw.RestartBrowser()
+			if err != nil {
+				broadcastErr := fmt.Errorf("issue encountered with PNG exporter, stopping")
+				w.ms.Log.Error.Print(broadcastErr)
+				w.broadcast(&compileResult{
+					Err: broadcastErr.Error(),
+				})
+				return fmt.Errorf("Playwright could not be restarted: %w", err)
```

Comment:
This loop is only supposed to return if the context is cancelled. So remove this line.

Reply:
then it'd proceed to compile?

Reply:
Ah right, then make it `continue`.

Reply:
If we do continue, will it continue to run even though they're trying export to ".png" but the png Exporter has died?

Reply:
no ctx.err will be hit